metricingester/aggingestor.go

```diff
@@ -0,0 +1,142 @@
+package metricingester
+
+import (
+	"context"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/stripe/veneur/trace"
+
+	"github.com/stripe/veneur/samplers"
+)
+
+type AggregatingIngestor struct {
+	workers []aggWorker
+	flusher flusher
+	ticker  *time.Ticker
+	tickerC <-chan time.Time
+	quit    chan struct{}
+	logger  *logrus.Logger
+	tc      *trace.Client
+}
+
+type flusher interface {
+	Flush(ctx context.Context, envelope samplerEnvelope)
+}
+
+type ingesterOption func(AggregatingIngestor) AggregatingIngestor
+
+// Override the ticker channel that triggers flushing. Useful for testing.
+func OptFlushChan(tckr <-chan time.Time) ingesterOption {
+	return func(option AggregatingIngestor) AggregatingIngestor {
+		option.tickerC = tckr
+		return option
+	}
+}
+
+func OptLogger(logger *logrus.Logger) ingesterOption {
```

Comment:
Believe this is unused, but should be used in server.go, like the `OptTraceClient` function below.

Reply:
I'm not sure what happens when you use a nil `logger` either... does it work, or does it die?